packages/flutter_tools/lib/src/android/gradle_errors.dart

```diff
@@ -0,0 +1,299 @@
+      '${licenseMatch.group(1)}\n\n'
+      'To resolve this, please run the following command in a Terminal:\n'
+      'flutter doctor --android-licenses'
+    );
+    BuildEvent('license-not-accepted').send();
+    return GradleBuildStatus.EXIT;
+  },
+);
+
+final RegExp _undefinedTaskPattern = RegExp(r'Task .+ not found in root project.');
+
+final RegExp _assembleTaskPattern = RegExp(r'assemble(\S+)');
+
+/// Handler when a flavor is undefined.
+@visibleForTesting
+final GradleHandledError flavorUndefinedHandler = GradleHandledError(
+  test: (String line) {
+    return _undefinedTaskPattern.hasMatch(line);
+  },
+  handler: ({
+    String line,
+    FlutterProject project,
+    bool usesAndroidX,
+    bool shouldBuildPluginAsAar,
+  }) async {
+    final RunResult tasksRunResult = await processUtils.run(
+      <String>[
+        gradleUtils.getExecutable(project),
+        'app:tasks' ,
+        '--all',
+        '--console=auto',
+      ],
+      throwOnError: true,
+      workingDirectory: project.android.hostAppGradleRoot.path,
+      environment: gradleEnv,
+    );
+    // Extract build types and product flavors.
+    final Set<String> variants = <String>{};
+    for (String task in tasksRunResult.stdout.split('\n')) {
+      final Match match = _assembleTaskPattern.matchAsPrefix(task);
+      if (match != null) {
+        final String variant = match.group(1).toLowerCase();
+        if (!variant.endsWith('test')) {
+          variants.add(variant);
+        }
+      }
+    }
+    final Set<String> productFlavors = <String>{};
+    for (final String variant1 in variants) {
+      for (final String variant2 in variants) {
+        if (variant2.startsWith(variant1) && variant2 != variant1) {
+          final String buildType = variant2.substring(variant1.length);
+          if (variants.contains(buildType)) {
+            productFlavors.add(variant1);
+          }
+        }
+      }
+    }
+    printStatus(
+      '\n$warningMark  Gradle project does not define a task suitable '
+      'for the requested build.'
+    );
+    if (productFlavors.isEmpty) {
+      printStatus(
+        'The android/app/build.gradle file does not define '
+        'any custom product flavors. '
+        'You cannot use the --flavor option.'
+      );
+    } else {
+      printStatus(
+        'The android/app/build.gradle file defines product '
+        'flavors: ${productFlavors.join(', ')} '
+        'You must specify a --flavor option to select one of them.'
+      );
+    }
+    BuildEvent('flavor-undefined').send();
+    return GradleBuildStatus.EXIT;
+  },
+);
+
+/// The status of the Gradle build.
+enum GradleBuildStatus{
+  /// The tool cannot recover from the failure and should exit.
+  EXIT,
+  /// The tool can retry the exact same build.
+  RETRY,
+  /// The tool can build the plugins as AAR and retry the build.
+  RETRY_WITH_AAR_PLUGINS,
+}
+
+/// A Gradle handled error.
+///
+/// When the [test] function returns [true], [handler] is called.
```

Comment:
This is not an accurate description of this class. Nothing in this class actually calls handler (or test, for that matter).